src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -100,10 +102,16 @@ public function get( $url ) {
 			$response    = $this->remote_request->get( $url );
 			$expiry      = $this->get_expiry_time( $response );
 			$cached_data = new CachedData( $response->getBody(), $expiry );
-			set_transient( $cache_key, serialize( $cached_data ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+			$headers     = $response->getHeaders();
+
+			// Only store response if it was successful.
+			$status = $response->getStatusCode();
+			if ( $status >= 200 && $status < 300 ) {
+				set_transient( $cache_key, serialize( $cached_data ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+			}
```

Comment:
What about if the request fails with every request? Doesn't this mean an HTTP request will be issued for every page load? Before this PR would cache failures for a month the same as successes. But #4293 notes:

> * Fetches that result in failures should be cached for a `DAY_IN_SECONDS` rather than `MONTH_IN_SECONDS`.

Is that not accounted for?

Reply:
What if `$expiry` were split into two separate arguments, one for `$expiry_success` and another for `$expiry_failure`?

One more thing: what if the `max-age` in the `Cache-Control` is set to zero or something very short? We should not honor such max-ages. In that case, what about making `$expiry` (or `$expiry_success`) into a `$min_expiry`? This could then be paired with the `$use_cache_control` to be whatever the `max` value is.

I don't think we should allow for an expiry of less than a certain amount of time. This would go hand-in-hand with #2092, where if a `max-age` is `0` or very short, and the stylesheet is responding with random CSS, then it will cause transients to fill up.

Reply:
Right now it caches `DAY_IN_SECONDS` by default, _unless_ a non-zero `max-age` is provided via `cache-control`.

However, I can go with a default and a min expiry instead, to have a few basic guarantees.

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/4404/commits/b8e10da7b4adc2e8a03b458fa546c14318e1ebe9